src/txrebroadcast.cpp

```diff
@@ -20,10 +21,15 @@ std::vector<TxIds> TxRebroadcastHandler::GetRebroadcastTransactions()
 {
     std::vector<TxIds> rebroadcast_txs;
 
+    // If there has not been a cache run since the last block, the fee rate
+    // condition will not filter out any transactions, so skip this run.
+    if (m_tip_at_cache_time == m_chainman.ActiveChain().Tip()) return rebroadcast_txs;
```

Comment:
you're right, this comment is really confusing. updating to: 
```
    // If the cache has run since we received the last block, the fee rate
    // condition will not filter out any transactions, so skip this run.
```

And here's some more explanation: 

Let's refer to the fee rate for a txn to be in the top `MAX_REBROADCAST_WEIGHT` as top-of-mempool fee rate. The only event that has the ability to decrease the top-of-mempool fee rate is a block being mined. Otherwise, as transactions enter the mempool, the top-of-mempool fee rate either stays the same or increases. 

If the top-of-mempool fee rate is only increasing over time, the fee rate cache would not help filter any transactions out. 

Our desired order is: cache | block | rebroadcast, with the delta between cache & block to be as small as possible. 

The order we want to skip is: block | cache | rebroadcast, because then the fee rate filter will be meaningless, and would simply select the _next highest_ transactions to rebroadcast (which we logically wouldn't expect to be mined yet). 

Does this make sense? Does the update to the comment help clarify? I'm not sure the best way to capture this in the code, so open to suggestions.